internal/service/lightsail/container_service_test.go

```diff
@@ -0,0 +1,1306 @@
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "new_value1"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckContainerServiceDestroy(s *terraform.State) error {
+	conn := acctest.Provider.Meta().(*conns.AWSClient).LightsailConn
+
+	for _, r := range s.RootModule().Resources {
+		if r.Type != "aws_lightsail_container_service" {
+			continue
+		}
+
+		input := lightsail.GetContainerServicesInput{
+			ServiceName: aws.String(r.Primary.ID),
+		}
+
+		_, err := conn.GetContainerServices(&input)
+		if err == nil {
+			return fmt.Errorf("container service still exists: %s", r.Primary.ID)
+		}
+
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == lightsail.ErrCodeNotFoundException {
+				return nil
+			}
+		}
+		return err
+	}
+
+	return nil
+}
+
+func testAccCheckContainerServiceExists(resourceName string, cs *lightsail.ContainerService) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not finding Lightsail Container Service (%s)", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no Lightsail Container Service ID is set")
+		}
+
+		conn := acctest.Provider.Meta().(*conns.AWSClient).LightsailConn
+		resp, err := conn.GetContainerServices(&lightsail.GetContainerServicesInput{
+			ServiceName: aws.String(rs.Primary.ID),
+		})
+		if err != nil {
+			return err
+		}
+		*cs = *resp.ContainerServices[0]
+
+		return nil
+	}
+}
+
+func testAccContainerServiceConfigBasic(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainer1(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+
+  deployment {
+    container {
+      container_name = "test1"
+      image = "amazon/amazon-lightsail:hello-world"
+    }
+  }
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainer2(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+
+  deployment {
+    container {
+      container_name = "test1"
+      image = "amazon/amazon-lightsail:hello-world"
+    }
+
+    container {
+      container_name = "test2"
+      image = "redis:latest"
+    }
+  }
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainer3(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_lightsail_container_service" "test" {
+  name = %q
+  power = "nano"
+  scale = 1
+
+  deployment {
+    container {
+      container_name = "test3"
+      image = "nginx:latest"
+    }
+  }
+}
+`, rName)
+}
+
+func testAccContainerServiceConfigDeploymentContainerEnvironment1(rName string) string {
```

Comment:
It would be nice to see this test `testAccContainerServiceConfigDeploymentContainerEnvironment1` and `testAccContainerServiceConfigDeploymentContainerEnvironment2` combined with an additional parameter for environment to reduce code duplication.

Reply:
https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/running-and-writing-acceptance-tests.md#resource-acceptance-testing says to write tests to be `Per Attribute`. So, I did not combine too much. Perhaps this can be future work..